shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/security/DatasetSecurityService.java

```diff
@@ -574,36 +574,70 @@ public boolean hasUpdateRightOnDataset(Dataset dataset, String rightStr) throws
     	}
     }
 
+	/**
+	 * Check the connected user has the given right for the given dataset acquisition.
+	 * If the study is updated, check the user has the given right in both former and new studies.
+	 *
+	 * @param datasetAcq the dataset acquisition
+	 * @param rightStr the right
+	 * @return true or false
+	 * @throws EntityNotFoundException
+	 */
+	public boolean hasUpdateRightOnDatasetAcquisition(DatasetAcquisition datasetAcq, String rightStr) throws EntityNotFoundException {
```

Comment:
This is a very good question. I've worked on these because this was not working, but I didn't really understood the logic here.

Reply:
Hi Youenn, it would be great to explore on this please? When the studyId was different (source of the problem), there might be another issue beforehand during the creation of the exam with the frontend (see https://github.com/fli-iam/shanoir-ng/blob/b30c71fd3e8e831f2caee992e1492e68b487fde5/shanoir-ng-import/src/main/java/org/shanoir/ng/importer/bids/BidsImporterApiController.java#L106). I think your change solved the problem, but probably not the origin. Normally, when the exam is correctly created, the studyId must be the same. The user should only be able to create the exam in his studies. Maybe something is wrong with the rabbitmq message to ms datasets to create the correct exam?

Reply:
Hi @michaelkain,
You're right, checking that studyId did not change make no sense here, as it can't happen when updating an acquisition.

Shouldn't we just test rights on examination, as for an acquisition creation ?

Reply:
Done.